cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -576,8 +577,10 @@ private Object convertValue(TypeToken<?> fieldType, String value) throws Excepti
         } catch (InvocationTargetException e) {
           if (e.getCause() instanceof NumberFormatException) {
             // if exception is due to wrong value for integer/double conversion
-            throw new InvalidPluginConfigException(String.format("valueOf operation on %s failed", value),
-                                                   e.getCause());
+            String errorMessage = Strings.isNullOrEmpty(value) ?
+              String.format("Value of field %s is null or empty. It should be integer or double", name) :
```

Comment:
can't this be other number types like float and long? Maybe it would be better to say the field could not be parsed as a number?